daemon/cmd/healthz.go

```diff
@@ -0,0 +1,55 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"net/http"
+
+	"github.com/cilium/cilium/api/v1/models"
+)
+
+// startAgentHealthzHTTPService registers a handler function for the /healthz
+// status HTTP endpoint exposed on addr. This endpoint reports the agent health
+// status and is equivalent to what the `cilium status --brief` CLI tool reports.
+func (d *Daemon) startAgentHealthzHTTPService(addr string) {
+	http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I believe you meant to use http.Handle**r**Func here and pass it to `srv` as a handler. `http.HandleFunc` will register a global handler (for use with `http.ListenAndServe`) and may panic if other code registers the `/healthz` pattern.

Reply:
Oops, well spotted. I initially was using `http.ListenAndServe` and forgot to convert this to `http.HandlerFunc` when changing to `srv.ListenAndServe`.